components/src/dynamo/profiler/utils/dgdr_v1beta1_types.py

```
@@ -0,0 +1,370 @@
+            and self.planner
+            and (
+                self.planner.pre_deployment_sweeping_mode is None
+                or self.planner.pre_deployment_sweeping_mode
+                == PlannerPreDeploymentSweepMode.None_
+            )
+        ):
+            raise ValueError(
+                "pre_deployment_sweeping_mode cannot be 'none' when mocker is enabled. "
+                "Mocker backend requires pre-deployment sweeping to generate simulated "
+                "performance profiles."
+            )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)